### ISSUE TITLE
NACM blocks libyang context modules

### ISSUE THREAD

**Issue by syyyr** [CESNET/netopeer2#841]
Hi, I have found that the NACM config has an effect on what modules get filled into libyang context. For example, if I have module named `example-schema` installed and NACM with the default configuration:
```cpp
#include <cassert>
#include <libnetconf2/session_client.h>
#include <sysrepo-cpp/Session.hpp>
#include <memory>

int main(int argc, char* argv[])
{
    auto SR_conn = std::make_shared<sysrepo::Connection>();
    auto SR_sess = std::make_shared<sysrepo::Session>(SR_conn);
    auto SR_ctx = SR_sess->get_context();
    assert(SR_ctx->get_module("example-schema"));

    auto NC_sess = nc_connect_unix("/opt/cesnet/Netopeer2/netopeer2-server.sock", nullptr);
    auto NC_ctx = nc_session_get_ctx(NC_sess);
    assert(ly_ctx_get_module(NC_ctx, "example-schema", nullptr, 0));
}
```
this program crashes on the second assert. If I disable NACM, I can see the module again.

Also, a side note: NACM really slows `nc_connect_unix`. With it disabled, the program runs for `0m0.133s`, with it enabled, it runs for `0m2.225s` (but I haven't profiled yet, sorry).

**Comment by michalvasko** [comment]
I did not manage to replicate this but it took me a few tries until I actually connected and did not get `Permission denied` error. So it may be specific for a certain setup, I was running the server as `root`.

As for the inefficiency, there is definitely some room for improvement (but do not expect miracles, generally, NACM must check each node individually) but I was not able to work on it yet.
